CefSharp.Core/Internals/ClientAdapter.cpp

```diff
@@ -290,6 +307,28 @@ namespace CefSharp
             }			
         }
 
+        void ClientAdapter::OnResourceRedirect(CefRefPtr<CefBrowser> browser, CefRefPtr<CefFrame> frame, const CefString& oldUrl, CefString& newUrl)
+        {
+            IRequestHandler^ handler = _browserControl->RequestHandler;
+            if (handler != nullptr)
+            {
+                auto managedNewUrl = StringUtils::ToClr(newUrl);
+                handler->OnResourceRedirect(_browserControl, frame->IsMain(), StringUtils::ToClr(oldUrl), managedNewUrl);
+
+                //TODO: Determine if there is a better way to pass the reference in rather than reassigning the string
+                newUrl = StringUtils::ToNative(managedNewUrl);
```

Comment:
You might be able to make the statement conditional on if the strings are actually different I suppose, but that degrades into a string comparison. Given that the Windows heaps are of the very low fragmenting variety, and any tiny allocations this might add to the pain are probably dwarfed by CEF doing its browser UI thing, I wouldn't worry about it and just leave it like it is.